processor/src/main/java/org/mapstruct/ap/util/Message.java

```diff
@@ -88,6 +88,12 @@
     RETRIEVAL_PRIMITIVE_RETURN( "Can't generate mapping method with primitive return type." ),
     RETRIEVAL_ENUM_TO_NON_ENUM( "Can't generate mapping method from enum type to non-enum type." ),
     RETRIEVAL_NON_ENUM_TO_ENUM( "Can't generate mapping method from non-enum type to enum type." ),
+    RETRIEVAL_TYPE_VAR_SOURCE( "Can't generate mapping method for a generic type variable source." ),
```

Comment:
generics come basically in 2 forms: parameterized (like <T extends Something> on method / class level) and wildcards (the question mark). A generic type variable is the first one.  Checkout http://docs.oracle.com/javase/7/docs/api/javax/lang/model/type/TypeKind.html